### PR TITLE
[runtime] Set exit code on unhandled exception even if a managed handler is present

### PR DESCRIPTION
**Pull request by ludovic-henry**
In case we would add a AppDomain.CurrentDomain.UnhandledException callback, we would not exit the VM in case of unhandled exception.
This goes against the expected behaviour, as we should indeed call these managed callbacks in all cases, but they should not stop the VM from exiting.

Fixes https://bugzilla.xamarin.com/show_bug.cgi?id=39077

### CODE REVIEW

**Comment by ludovic-henry** [comment]
@monojenkins merge

File: mono/metadata/object.c [hunk 1]
```diff
@@ -4326,6 +4319,13 @@ mono_unhandled_exception (MonoObject *exc)
 		if (current_appdomain_delegate)
 			call_unhandled_exception_delegate (current_domain, current_appdomain_delegate, exc);
```
Comment (akoeplinger) [inline mono/metadata/object.c]: Unrelated to this PR, but this seems in the wrong order. According to [MSDN](https://msdn.microsoft.com/de-de/library/system.appdomain.unhandledexception(v=vs.110).aspx):

> If the UnhandledException event is handled in the default application domain, it is raised there for any unhandled exception in any thread, no matter what application domain the thread started in.**If the thread started in an application domain that has an event handler for UnhandledException, the event is raised in that application domain**.If that application domain is not the default application domain, and there is also an event handler in the default application domain, the event is raised in both application domains.

We should be calling the current_appdomain_delegate before root_appdomain_delegate

**Comment by akoeplinger** [comment]
@ludovic-henry can you please add a testcase for this?

File: mono/metadata/object.c (hunk 1 above)
Comment (akoeplinger) [inline mono/metadata/object.c]: Well after reading it again, it doesn't strictly specify the order. nevermind :smile:

**Comment by ludovic-henry** [comment]
build

**Comment by ludovic-henry** [comment]
build

**Comment by akoeplinger** [comment]
Marek removed the test, so this needs a rebase.

**Comment by ludovic-henry** [comment]
@monojenkins merge

**Pull request merged**